### PR TITLE
[Merged by Bors] - feat: port data.rat.init

### PR DESCRIPTION
**Pull request by mcdoll**
All the constructions are in `Std.Data.Rat`, we only add the notation.

### CODE REVIEW

**Comment by mcdoll** [comment]
how do I add nolints?

**Comment by digama0** [comment]
You should just add a module doc. It can be a one-liner if you don't have anything to say.

File: Mathlib/Init/Data/Rat.lean [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import Std.Data.Rat
+
+/-!
+# Notation for the rational numbers -/
+
+namespace Rat
+
+/-- Write the rational numbers as `ℚ`. -/
+notation "ℚ" => Rat
```
Comment (digama0) [inline Mathlib/Init/Data/Rat.lean:15]:
```suggestion
@[inherit_doc] notation "ℚ" => Rat
```
we want people hovering on `ℚ` to see the documentation about `Rat`, not the documentation about the notation.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**